packages/node-addon-examples/src/index.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+
+function assertLogs(cb: () => void, expectedMessages: string[]) {
+  const errors: Error[] = [];
+  // Spying on the console.log function, as the examples don't assert anything themselves
+  const originalLog = console.log;
+  console.log = (message: string, ...args: unknown[]) => {
+    const nextMessage = expectedMessages.shift();
+    const combinedMessage = [message, ...args].map(String).join(" ");
+    if (nextMessage !== combinedMessage) {
+      errors.push(new Error(`Unexpected log message '${combinedMessage}'`));
+    }
+  };
+  try {
+    cb();
+    if (expectedMessages.length > 0) {
+      errors.push(
+        new Error(`Missing expected message(s): ${expectedMessages.join(", ")}`)
+      );
+    }
+  } finally {
+    console.log = originalLog;
+  }
+  // Throw and first error
+  const [firstError] = errors;
+  if (firstError) {
+    throw firstError;
+  }
+}
+
+export const suites: Record<string, Record<string, () => void>> = {
```

Comment:
I wasn't clear how/when this test file would be run, and whether it'd be in Node.js or React Native. May be worth adding some brief docs in `packages/node-addon-examples/README.md` and `apps/test-app/README.md`.

Reply:
I revisited the [packages/node-addon-examples/README.md](https://github.com/callstackincubator/react-native-node-api/blob/main/packages/node-addon-examples/README.md).